core/storage/user/gae_models.py

```diff
@@ -97,6 +97,61 @@ def get_by_normalized_username(cls, normalized_username):
             cls.normalized_username == normalized_username).get()
 
 
+class ActivitiesCompletedByLearnerModel(base_models.BaseModel):
+    """Keeps track of all the explorations and collections completed by the
+    learner.
+
+    Instances of this class are keyed by the user id.
+    """
+    # IDs of all the explorations completed by the user.
+    completed_exp_ids = ndb.StringProperty(repeated=True, indexed=True)
+    # IDs of all the collections completed by the user.
+    completed_collection_ids = ndb.StringProperty(repeated=True, indexed=True)
+
+
+class IncompleteCollectionsModel(base_models.BaseModel):
+    """Keeps track of all the collections currently being completed by the
+    learner.
+
+    Instances of this class are keyed by the user id.
+    """
+    incomplete_collection_ids = ndb.StringProperty(
```

Comment:
Do we know for certain that this isn't going to turn into a list-of-blobs property in the future, similar to IncompleteExplorationsModel? If not, see comment there for a possible way to mitigate this.